### PR TITLE
feat: propagate t4_dataset_id through bag_metadata

### PR DESCRIPTION
**Pull request by xtk8532704**
## What this PR does

Propagates upstream T4 dataset identity through the C++ rosbag → NPZ conversion path so converted datasets can carry `t4_dataset_id` / `t4_dataset_version_id` from `log_file_info.json` into frame / sidecar metadata.

This pairs with the meta-repo Evaluator download flow (`download_t4datasets_from_evaluator.py`), which already writes those fields into bag metadata JSON. Without this PR, the converter ignored them.

### Changes

- `BagMetadata` gains `t4_dataset_id` and `t4_dataset_version_id`.
- `load_bag_metadata` / `write_bag_metadata` read and write the fields (`get_string_or_empty` → missing keys become `""`).
- Unit tests: populated path (`T4DatasetFieldsAreLoaded`, `ValidJsonAllFields`) and empty / missing-file paths.

## Testing

- [ ] `test_bag_metadata` / `test_frame_writer` C++ unit tests pass
- [ ] Convert a bag whose `log_file_info.json` has T4 IDs and confirm they appear in output metadata

### CODE REVIEW

File: cpp_tools/src/autoware_diffusion_planner_tools/test/test_bag_metadata.cpp [hunk 1]
```diff
@@ -59,6 +59,8 @@ TEST_F(BagMetadataTest, ValidJsonAllFields)
   EXPECT_EQ(meta.map_version_id, "1847-20250819054643702212");
   EXPECT_EQ(meta.date, "2026-01-16");
   EXPECT_EQ(meta.bag_time, "10-03-35");
+  EXPECT_EQ(meta.t4_dataset_id, "");
```
Comment (danielsanchezaran) [inline cpp_tools/src/autoware_diffusion_planner_tools/test/test_bag_metadata.cpp]: Nit: `ValidJsonAllFields` now asserts the t4 fields are empty, i.e. the "all fields" fixture doesn't actually contain them despite the test name. Not a correctness problem (the populated path is covered by `T4DatasetFieldsAreLoaded`), but either add the two keys to this fixture or rename it so "all fields" isn't misleading.

**Review by danielsanchezaran — COMMENTED** [review]
LGTM on the T4 dataset metadata. Verified: `load_bag_metadata` reads both new fields via `get_string_or_empty`, so a missing/non-string key safely yields `""` (no throw); `write_bag_metadata` round-trips them; struct field order is consistent with the aggregate initializer in `test_frame_writer.cpp`; and `T4DatasetFieldsAreLoaded` covers the populated path while the existing missing-field/missing-file tests cover the empty path. One naming nit inline.

File: cpp_tools/src/autoware_diffusion_planner_tools/test/test_bag_metadata.cpp (hunk 1 above)
Comment (xtk8532704) [reply cpp_tools/src/autoware_diffusion_planner_tools/test/test_bag_metadata.cpp]: Thank you, fixed in [0d20bef](https://github.com/tier4/Diffusion-Planner/pull/278/commits/0d20befd6382459e2cb80138f5bae53f5b57b4b9)

**Review by danielsanchezaran — COMMENTED** [review]
Re-reviewed the follow-up commit after the split. The `ValidJsonAllFields` fixture now includes and asserts both populated T4 dataset fields, resolving the only earlier nit. The metadata change is isolated from reproducer/mining/evaluation behavior; its only workflow effect is that future C++ conversions preserve upstream dataset identity in emitted sidecars. The PR is already merged and its pre-commit check passed.
